### DISCUSSION TITLE
counting fruit effectively and allocating specific data

### DISCUSSION THREAD

**Discussion by xquantize** [ultralytics/yolov5#8135] [Q&A]
using deep sort to track fruit, wanting to know if there are better options to ensure a fruit rolling across a conveyor is tracked continuously as had a few incidents where the fruit gets slightly covered and causes the id to change.

also seeking advice on whats the best method to allocate information to the specific fruit, such as fruit id 1, size is xx for example.

**Comment by glenn-jocher** [comment]
@Dxrkstar 👋 Hello! Thanks for asking about object **tracking** in computer vision. YOLOv5 🚀 is an object **detector** that detect, localizes and classifies objects in a single image. It does not connect objects across multiple images, for this you need a tracking solution. A few possible tracking solutions are:

- [Extended Kalman Filter (EKF)](https://en.wikipedia.org/wiki/Kalman_filter)
- [Kanade-Lukas-Tomasi (KLT) Feature Tracker](https://en.wikipedia.org/wiki/Kanade%E2%80%93Lucas%E2%80%93Tomasi_feature_tracker): 
- [YOLOv5 DeepSort Tracker](https://github.com/mikel-brostrom/Yolov5_DeepSort_Pytorch) 
- [Apple iOS VNTrackObjectRequest](https://developer.apple.com/documentation/vision/tracking_multiple_objects_or_rectangles_in_video)

Good luck 🍀 and let us know if you have any other questions!

**Reply by xquantize** [reply]
Thanks for the reply!

Do you know what the best way is to implement one of these trackers with YOLOv5 outputs? Or what from YOLO will I feed into them essentially in order to track specific detections.

**Reply by glenn-jocher** [reply]
@Dxrkstar implementation is up to you. They require an initial bounding box from YOLO to start tracking.
